boot/bootchain_test.go

```diff
@@ -987,3 +1004,169 @@ func (s *sealSuite) TestPredictableBootChainsSortOrder(c *C) {
 		},
 	})
 }
+
+func printChain(c *C, chain *secboot.LoadChain, prefix string) {
+	c.Logf("%v %v", prefix, chain.BootFile)
+	for _, n := range chain.Next {
+		printChain(c, n, prefix+"-")
+	}
+}
+
+// cPath returns a path under boot assets cache directory
+func cPath(p string) string {
+	return filepath.Join(dirs.SnapBootAssetsDir, p)
+}
+
+// nbf is bootloader.NewBootFile but shorter
+func nbf(snap, path, role string) bootloader.BootFile {
+	return bootloader.NewBootFile(snap, path, bootloader.Role(role))
+}
+
+func (s *bootchainSuite) TestBootAssetsToLoadChainTrivialKernel(c *C) {
+	kbl := bootloader.NewBootFile("pc-kernel", "kernel.efi", bootloader.Role("run"))
+
+	chains, err := boot.BootAssetsToLoadChains(nil, kbl, nil)
+	c.Assert(err, IsNil)
+
+	c.Check(chains, DeepEquals, []*secboot.LoadChain{
+		secboot.NewLoadChain(nbf("pc-kernel", "kernel.efi", "run")),
+	})
+}
+
+func (s *bootchainSuite) TestBootAssetsToLoadChainErr(c *C) {
+	kbl := bootloader.NewBootFile("pc-kernel", "kernel.efi", bootloader.Role("run"))
+
+	assets := []boot.BootAsset{
+		{Name: "shim", Hashes: []string{"hash0"}, Role: "recovery"},
+		{Name: "loader-recovery", Hashes: []string{"hash0"}, Role: "recovery"},
+		{Name: "loader-run", Hashes: []string{"hash0"}, Role: "run"},
```

Comment:
It doesn't make any difference here, but the run mode Role as defined in bootloader is "run-mode".